apps/src/ni_linemod.cpp

```diff
@@ -73,491 +73,527 @@ using PointT = PointXYZRGBA;
-      for (const int &index : points_above_plane->indices)
-        scene->points[index].label = 1;
-      euclidean_cluster_comparator->setLabels (scene);
-
-      OrganizedConnectedComponentSegmentation<PointT, Label> euclidean_segmentation (euclidean_cluster_comparator);
-      euclidean_segmentation.setInputCloud (cloud);
-
-      PointCloud<Label> euclidean_labels;
-      std::vector<PointIndices> euclidean_label_indices;
-      euclidean_segmentation.segment (euclidean_labels, euclidean_label_indices);
-
-      // For each cluster found
-      bool cluster_found = false;
-      for (const auto &euclidean_label_index : euclidean_label_indices)
-      {
-        if (cluster_found)
-          break;
-        // Check if the point that we picked belongs to it
-        for (std::size_t j = 0; j < euclidean_label_index.indices.size (); ++j)
-        {
-          if (picked_idx != euclidean_label_index.indices[j])
-            continue;
-          //pcl::PointCloud<PointT> cluster;
-          pcl::copyPointCloud (*cloud, euclidean_label_index.indices, object);
-          cluster_found = true;
-          break;
-          //object_indices = euclidean_label_indices[i].indices;
-          //clusters.push_back (cluster);
-        }
-      }
+    // Else, draw it on screen
+    // cloud_viewer_.addPolygon (region, 1.0, 0.0, 0.0, "region");
+    // cloud_viewer_.setShapeRenderingProperties
+    // (visualization::PCL_VISUALIZER_LINE_WIDTH, 10, "region");
+
+    PlanarRegion<PointT> refined_region;
+    pcl::approximatePolygon(region, refined_region, 0.01, false, true);
+    PCL_INFO("Planar region: %lu points initial, %lu points after refinement.\n",
+             region.getContour().size(),
+             refined_region.getContour().size());
+    cloud_viewer_.addPolygon(refined_region, 0.0, 0.0, 1.0, "refined_region");
+    cloud_viewer_.setShapeRenderingProperties(
+        visualization::PCL_VISUALIZER_LINE_WIDTH, 10, "refined_region");
+
+    // Draw in image space
+    image_viewer_.addPlanarPolygon(
+        search_.getInputCloud(), refined_region, 0.0, 0.0, 1.0, "refined_region", 1.0);
+
+    // If no object could be determined, exit
+    if (!object) {
+      PCL_ERROR("No object detected. Please select another point or relax the "
+                "thresholds and continue.\n");
+      return;
     }
+    // Visualize the object in 3D...
+    visualization::PointCloudColorHandlerCustom<PointT> red(object, 255, 0, 0);
+    if (!cloud_viewer_.updatePointCloud(object, red, "object"))
+      cloud_viewer_.addPointCloud(object, red, "object");
+    // ...and 2D
+    image_viewer_.removeLayer("object");
+    image_viewer_.addMask(search_.getInputCloud(), *object, "object");
+
+    // Compute the min/max of the object
+    PointT min_pt, max_pt;
+    getMinMax3D(*object, min_pt, max_pt);
+    stringstream ss2;
+    ss2 << "cube_" << idx;
+    // Visualize the bounding box in 3D...
+    cloud_viewer_.addCube(min_pt.x,
```

Comment:
Maybe we should switch format off for this; looks unwieldy

Reply:
It looks a big ugly - yes. But it will be also looks confusing why this line can bypass clang-format and all other not (e.g. this below). So I think it is better to let this line as it is now.